linera-views/src/set_view.rs

```diff
@@ -361,28 +330,11 @@ where
     type Hasher = sha3::Sha3_256;
 
     async fn hash_mut(&mut self) -> Result<<Self::Hasher as Hasher>::Output, ViewError> {
-        let hash = *self.hash.get_mut();
-        match hash {
-            Some(hash) => Ok(hash),
-            None => {
-                let new_hash = self.compute_hash().await?;
-                let hash = self.hash.get_mut();
-                *hash = Some(new_hash);
-                Ok(new_hash)
-            }
-        }
+        self.compute_hash().await
```

Comment:
Maybe just call `hash` and move `compute_hash` to `hash`?

Reply:
Yes, it can be done. Except for `register_view`, `collection_view`, and `reentrant_collection_view`.